### PR TITLE
Plane: Soaring: Allow pilot to commanded exit from thermalling via RC

### PR DESCRIPTION
**Pull request by samuelctabor**
This is a good idea from @Hwurzburg.

Right now we have an RC option for setting soaring behaviour. Switching between mid and high switch positions does nothing while the plane is in thermalling state in LOITER mode.

This PR allows using the switch (usually toggling to the other position and back) to command ending thermalling and restoring previous mode. This avoids the pilot needing to use the mode switch and it also allows the usual attempt to align the heading before actually changing mode.

### CODE REVIEW

File: libraries/AP_Soaring/AP_Soaring.h [hunk 1]
```diff
@@ -81,7 +81,8 @@ class SoaringController {
         THERMAL_WEAK,
         ALT_LOST,
         DRIFT_EXCEEDED,
-        GOOD_TO_KEEP_LOITERING
+        GOOD_TO_KEEP_LOITERING,
+        EXIT_COMMANDED
```
Comment (peterbarker) [inline libraries/AP_Soaring/AP_Soaring.h:85]: trailing comma here would be nice.  Simplifies the deltas.

**Comment by peterbarker** [comment]
How was this tested?

File: libraries/AP_Soaring/AP_Soaring.h (hunk 1 above)
Comment (samuelctabor) [reply libraries/AP_Soaring/AP_Soaring.h]: Thanks, fixed

**Comment by samuelctabor** [comment]
> How was this tested?

Tested in flight and in SITL.

**Review by magicrub — APPROVED** [review]

**Comment by magicrub** [comment]
This LGTM. Ready to merge but will give @Hwurzburg a couple days to take a peek before I merge it

**Comment by Hwurzburg** [comment]
so I  am in LOITER with switch high...move to mid and back will exit thermal as if it ended due to low lift and return to mode (AUTO/CRUISE/FBWB) with SOARING still fully active for next thermal, correct? 
what if I was in LOITER while in mid position (manual thermal entry), does toggling to high (LOITER enable) mode and back to mid, exit the LOITER?

**Comment by samuelctabor** [comment]
> so I am in LOITER with switch high...move to mid and back will exit thermal as if it ended due to low lift and return to mode (AUTO/CRUISE/FBWB) with SOARING still fully active for next thermal, correct?
> what if I was in LOITER while in mid position (manual thermal entry), does toggling to high (LOITER enable) mode and back to mid, exit the LOITER?

Correct.
Yes, you can either toggle from mid->high->mid, or high->mid->high, so the switch is back in the original position

**Comment by Hwurzburg** [comment]
LGTM

**Pull request merged**